examples/idcomp.ctt

```diff
@@ -0,0 +1,253 @@
+--
+--                 refl
+--             a -------> a
+--             |          |
+--       refId |          | wrefId
+--             v   refl   v
+--             a -------> a
+--             |          |
+--      wrefId |          | refId
+--             v   refl   v
+--             a -------> a
+--
+-- We can now compose in the vertical direction to get a square:
+--
+--                 refl
+--             a -------> a
+--  refId      |          | wrefId
+--   . wrefId  |          |   . refId
+--             v   refl   v
+--             a -------> a
+--
+-- where p . q is the composition of p followed by q.
+
+-- If . is the built-in composition then both sides are wrefId
+-- and we get the following:
+
+square (A : U)(a : A) : Path (Id A a a) (wrefId A a) (wrefId A a) =
+  <i> idComp A a a a (s2w A a @ i) (w2s A a @ i)
+
+-- Gives:
+--         <i> idC (<j> comp (<_> A) a
+--               [ (i = 0) -> <_> a
+--               , (i = 1) -> <_> a
+-- 		 , (j = 0) -> <_> a
+-- 		 , (j = 1) -> <_> a ]) []
+
+-- Using the defined composition the left hand side of the square
+-- is not definitionally equal to wref and so results in a path with
+-- a starting point containing a trvial comp.
+
+wrefCompId (A : U)(a : A) : Id A a a =
+  idC (<i> comp (<_> A) a [(i=0) -> <_> a, (i=1) -> <_> a]) []
+
+square' (A : U)(a : A) : Path (Id A a a) (wrefCompId A a) (wrefId A a) =
+  <i> compId A a a a (s2w A a @ i) (w2s A a @ i)
+
+-- Gives:
+--         <i> idC (<j> comp (<_> A) a
+--               [ (i = 1) -> <_> a
+--               , (j = 0) -> <_> a
+--               , (j = 1) -> <_> a ]) []
+--
+-- Note that here we only have 3 out of 4 edges defined, as opposed to
+-- having all 4 defined previously.
+
+
+-- Consider the following type:
+data IdCompTest = a | b
+  | sq1 <i j> [ (i=0) -> a, (i=1) -> a, (j=0) -> a, (j=1) -> a ]
+  | sq2 <i j> [ (j=0) -> a, (j=1) -> b ]
+
+-- We can visualise sq1 and sq2 as two squares with a common
+-- face, like so:
+--
+--                 refl
+--             a -------> a
+--             |          |
+--       refl  |          | refl
+--             v   refl   v
+--             a -------> a
+--             |          |
+--             |          |
+--             v   refl   v
+--             b -------> b
+--
+-- Encode the top square as p1:
+
+p1 : Path (Id IdCompTest a a) (refId IdCompTest a) (refId IdCompTest a) =
+  <i> idC (<j> sq1 {IdCompTest} @ i @ j) [ (i=0) -> a, (i=1) -> a ]
+
+-- And the left and right sides of the bottom square as:
+
+id1 : Id IdCompTest a b = idC (<j> sq2 {IdCompTest} @ 0 @ j) []
+id2 : Id IdCompTest a b = idC (<j> sq2 {IdCompTest} @ 1 @ j) []
+
+-- The bottom square:
+
+p2 : Path (Id IdCompTest a b) id1 id2 =
+  <i> idC (<j> sq2 {IdCompTest} @ i @ j) []
+
+-- We can now compose in the vertical direction to get a square:
+--
+--                 refl
+--             a -------> a
+--   refId     |          |  refId
+--    . id1    |          |    . id2
```

Comment:
I am again nitpicking, but I have a feeling that you want the dot `.` before `id1` to be aligned with `f`.